packages/rspack/src/NativeWatchFileSystem.ts

```diff
@@ -0,0 +1,129 @@
+import * as binding from "@rspack/binding";
+import type Watchpack from "watchpack";
+import type {
+	FileSystemInfoEntry,
+	WatchFileSystem,
+	Watcher,
+	WatcherIncrementalDependencies
+} from "./util/fs";
+
+export default class NativeWatchFileSystem implements WatchFileSystem {
+	#inner: binding.NativeWatcher | undefined;
+
+	async watch(
```

Comment:
async with callback is weird, and it seems webpack's NodeWatchFileSystem is not async method, is the async really necessary?